test/integration/server.h

```diff
@@ -413,11 +413,11 @@ class IntegrationTestServer : public Logger::Loggable<Logger::Id::testing>,
              std::chrono::seconds drain_time, Server::DrainStrategy drain_strategy);
 
   void waitForCounterEq(const std::string& name, uint64_t value) override {
-    notifyingStatsAllocator().waitForCounterFromStringEq(name, value);
+    TestUtility::waitForCounterGe(statStore(), name, value, time_system_);
   }
 
   void waitForCounterGe(const std::string& name, uint64_t value) override {
-    notifyingStatsAllocator().waitForCounterFromStringGe(name, value);
+    TestUtility::waitForCounterGe(statStore(), name, value, time_system_);
```

Comment:
Somewhere here we should warn about the default timeout in the comments, because it's opaque that it might throw an exception.

I think it might make more sense to return an `AssertionResult` instead of throwing, but calls to these waitFor's will need to be wrapped with ASSERT_TRUE

Reply:
That makes sense.  Might be better to split this up into two functions, say `waitForCounterGe` and `waitForCounterGeTimeout` since there are dozens of calls to these across all the different tests.  The fuzz test is probably the only one that would expect these functions to timeout, say if there is a bug in xDS and the update doesn't properly go through, since mostly they're used to wait for the server to catch up to updates in integration tests.  What do you think?

Reply:
I think that without `ABSL_MUST_USE_RESULT` you can ignore the AssertionResult.

Reply:
Oh right sure, I thought you meant any calls to waitFor would need to be wrapped with ASSERT_TRUE, but I can just do it in the fuzzer.  I'll make that change, thanks!

Reply:
edit: I just moved the assert true to the server stub instead so that all calls will use it